sphinxcontrib/inmanta/api.py

```diff
@@ -435,62 +588,198 @@ def filter_func(name):
+            raise Exception(f"Could not find module {module_name} in {module_source_dir}.")
         mod, submodules = mod_data
 
-        module_filter = self.get_module_filter(None if not module_repo else os.path.join(module_repo, module_name))
+        module_filter = self.get_module_filter(
+            None
+            if not module_source_dir
+            else os.path.join(module_source_dir, module_name)
+        )
 
         for name in extra_modules:
-            extra_mod_data: Optional[Tuple[module.Module, List[str]]] = self._get_modules(module_repo, name)
+            extra_mod_data: Optional[Tuple[module.Module, List[str]]] = (
+                self._get_modules(module_source_dir, name)
+            )
             if extra_mod_data is not None:
                 submodules.extend(extra_mod_data[1])
 
         submodules = sorted([sm for sm in set(submodules) if module_filter(sm)])
         print("Selected sub-modules: " + ", ".join(submodules))
 
-        lines = self.emit_intro(mod, source_repo)
-        lines.extend(self.doc_compile(module_repo, mod.name, submodules))
+        lines = self.emit_intro(mod)
+        lines.extend(self.doc_compile(module_source_dir, mod.name, submodules))
         lines = [line for line in lines if line is not None]
         return "\n".join(lines)
 
-
 @click.command()
 @click.option(
-    "--module_repo",
+    "--module-sources",
     help=(
-        "The repo where all v1 modules are stored (local file). Ignored for v2 modules, which are always fetched from the"
-        " Python environment."
+        "The path to the local directory where all module sources (both v1 and v2) are stored."
     ),
 )
-@click.option("--module", help="The module to generate api docs for", required=True)
-@click.option("--extra-modules", "-m", help="Extra modules that should be loaded to render the docs", multiple=True)
-@click.option("--source-repo", help="The repo where the upstream source is located.")
-@click.option("--file", "-f", help="Save the generated result here.", required=True)
-def generate_api_doc(module_repo: Optional[str], module: str, extra_modules: Sequence[str], source_repo: str, file: str):
+@click.option(
+    "--module-name", help="The module to generate api docs for", required=True
+)
+@click.option(
+    "--extra-modules",
+    "-m",
+    help="Extra modules that should be loaded to render the docs",
+    multiple=True,
+)
+@click.option(
+    "--out-dir",
+    "-d",
+    help="Path to directory in which to put documentation.",
+    required=True,
+)
+def generate_module_doc(
+    module_sources: str,
+    module_name: str,
+    extra_modules: Sequence[str],
+    out_dir: str,
+):
+    """
+    Generate API documentation for a module. This command expects the module source to live in a directory
+    that has the module's name and that is in the <module_sources> directory.
+
+    The output of this command depends on whether the module contains
+    a README.md file or not:
```

Comment:
I wonder if we should add an option for the old behavior, in case users already have a `README.md` but are not ready to go to the new flow yet? e.g. `--ignore-readme` or `--autodoc-only`.

Reply:
Added the option and tested it manually